### PR TITLE
UCT/UD/MLX5: Set AV.GRH segment if needed.

### LINKED ISSUE #996: hang on gtest with ud_mlx5 on an ethernet port on CX4
There is a hang when running several gtest tests with ud_mlx5 on an Ethernet port on CX4.

To reproduce:
```
$./test/gtest/gtest --gtest_filter=ud_mlx5/test_uct_perf.envelope/*
[     INFO ] checking dc: yes
[     INFO ] checking dcx: yes
[     INFO ] checking ud: yes
[     INFO ] checking udx: yes
[     INFO ] checking udrc: yes
[     INFO ] checking cmrcx: yes
[     INFO ] checking shm: yes
[     INFO ] checking udrcx: yes
[     INFO ] checking ugni: no
[     INFO ] checking self: yes
[     INFO ] checking rc: yes
[     INFO ] checking rc_x: yes
[     INFO ] checking ud_mlx5: yes
[     INFO ] checking all: yes
[     INFO ] Using random seed of 32597
Note: Google Test filter = ud_mlx5/test_uct_perf.envelope/*
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from ud_mlx5/test_uct_perf
[ RUN      ] ud_mlx5/test_uct_perf.envelope/0
[     INFO ] Not enforcing performance on SandyBridge far socket
...hang...
```

ud and rc_mlx5 work:
```
alinas@hpc-test-node2 ~/workspace/tmp/ucx
$UCX_IB_ETH_PAUSE_ON=y ./test/gtest/gtest --gtest_filter=rc_mlx5/test_uct_perf.envelope/*
[     INFO ] checking dc: yes
[     INFO ] checking dcx: yes
[     INFO ] checking ud: yes
[     INFO ] checking udx: yes
[     INFO ] checking udrc: yes
[     INFO ] checking cmrcx: yes
[     INFO ] checking shm: yes
[     INFO ] checking udrcx: yes
[     INFO ] checking ugni: no
[     INFO ] checking self: yes
[     INFO ] checking rc: yes
[     INFO ] checking rc_x: yes
[     INFO ] checking ud_mlx5: yes
[     INFO ] checking all: yes
[     INFO ] Using random seed of 32112
Note: Google Test filter = rc_mlx5/test_uct_perf.envelope/*
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from rc_mlx5/test_uct_perf
[ RUN      ] rc_mlx5/test_uct_perf.envelope/0
[     INFO ] Not enforcing performance on SandyBridge far socket
[     INFO ] mlx5_1:1                am latency : 0.772 usec
[     INFO ] mlx5_1:1                   am rate : 6.239 Mpps
[     INFO ] mlx5_1:1                 am rate64 : 6.089 Mpps
[     INFO ] mlx5_1:1               am bcopy bw : 3182.375 MB/sec
[     INFO ] mlx5_1:1               am zcopy bw : 3020.934 MB/sec
[     INFO ] mlx5_1:1               put latency : 0.752 usec
[     INFO ] mlx5_1:1                  put rate : 4.037 Mpps
[     INFO ] mlx5_1:1              put bcopy bw : 4302.898 MB/sec
[     INFO ] mlx5_1:1              put zcopy bw : 2389.344 MB/sec
[     INFO ] mlx5_1:1               get latency : 1.467 usec
[     INFO ] mlx5_1:1        atomic add latency : 1.148 usec
[     INFO ] mlx5_1:1           atomic add rate : 0.627 Mpps
[     INFO ] mlx5_1:1       atomic fadd latency : 1.546 usec
[     INFO ] mlx5_1:1      atomic cswap latency : 1.569 usec
[     INFO ] mlx5_1:1       atomic swap latency : 1.591 usec
[       OK ] rc_mlx5/test_uct_perf.envelope/0 (7336 ms)
[ RUN      ] rc_mlx5/test_uct_perf.envelope/1
[     INFO ] Not enforcing performance on SandyBridge far socket
[     INFO ] mlx5_0:1                am latency : 0.737 usec
[     INFO ] mlx5_0:1                   am rate : 6.999 Mpps
[     INFO ] mlx5_0:1                 am rate64 : 6.227 Mpps
[     INFO ] mlx5_0:1               am bcopy bw : 3076.944 MB/sec
[     INFO ] mlx5_0:1               am zcopy bw : 2958.635 MB/sec
[     INFO ] mlx5_0:1               put latency : 0.706 usec
[     INFO ] mlx5_0:1                  put rate : 4.017 Mpps
[     INFO ] mlx5_0:1              put bcopy bw : 4550.800 MB/sec
[     INFO ] mlx5_0:1              put zcopy bw : 6529.128 MB/sec
[     INFO ] mlx5_0:1               get latency : 1.400 usec
[     INFO ] mlx5_0:1        atomic add latency : 1.101 usec
[     INFO ] mlx5_0:1           atomic add rate : 2.344 Mpps
[     INFO ] mlx5_0:1       atomic fadd latency : 1.454 usec
[     INFO ] mlx5_0:1      atomic cswap latency : 1.477 usec
[     INFO ] mlx5_0:1       atomic swap latency : 1.496 usec
[       OK ] rc_mlx5/test_uct_perf.envelope/1 (4819 ms)
[----------] 2 tests from rc_mlx5/test_uct_perf (12156 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (12156 ms total)
[  PASSED  ] 2 tests.
```

```
alinas@hpc-test-node2 ~/workspace/tmp/ucx
$./test/gtest/gtest --gtest_filter=ud/test_uct_perf.envelope/*
[     INFO ] checking dc: yes
[     INFO ] checking dcx: yes
[     INFO ] checking ud: yes
[     INFO ] checking udx: yes
[     INFO ] checking udrc: yes
[     INFO ] checking cmrcx: yes
[     INFO ] checking shm: yes
[     INFO ] checking udrcx: yes
[     INFO ] checking ugni: no
[     INFO ] checking self: yes
[     INFO ] checking rc: yes
[     INFO ] checking rc_x: yes
[     INFO ] checking ud_mlx5: yes
[     INFO ] checking all: yes
[     INFO ] Using random seed of 142
Note: Google Test filter = ud/test_uct_perf.envelope/*
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from ud/test_uct_perf
[ RUN      ] ud/test_uct_perf.envelope/0
[     INFO ] Not enforcing performance on SandyBridge far socket
[     INFO ] mlx5_1:1                am latency : 1.138 usec
[     INFO ] mlx5_1:1                   am rate : 2.024 Mpps
[     INFO ] mlx5_1:1                 am rate64 : 2.856 Mpps
[     INFO ] mlx5_1:1               am bcopy bw : 2322.376 MB/sec
[     INFO ] mlx5_1:1               am zcopy bw : 1401.015 MB/sec
[       OK ] ud/test_uct_perf.envelope/0 (3240 ms)
[ RUN      ] ud/test_uct_perf.envelope/1
[     INFO ] Not enforcing performance on SandyBridge far socket
[     INFO ] mlx5_0:1                am latency : 1.062 usec
[     INFO ] mlx5_0:1                   am rate : 3.043 Mpps
[     INFO ] mlx5_0:1                 am rate64 : 2.890 Mpps
[     INFO ] mlx5_0:1               am bcopy bw : 2267.674 MB/sec
[     INFO ] mlx5_0:1               am zcopy bw : 1362.044 MB/sec
[       OK ] ud/test_uct_perf.envelope/1 (3190 ms)
[----------] 2 tests from ud/test_uct_perf (6430 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (6430 ms total)
[  PASSED  ] 2 tests.
```

In this example mlx5_1:1 is an Eth port.
```
alinas@hpc-test-node2 ~/workspace/tmp/ucx
$ibv_devinfo
hca_id: mlx5_1
        transport:                      InfiniBand (0)
        fw_ver:                         12.14.1100
        node_guid:                      248a:0703:003f:1837
        sys_image_guid:                 248a:0703:003f:1836
        vendor_id:                      0x02c9
        vendor_part_id:                 4115
        hw_ver:                         0x0
        board_id:                       MT_2190110032
        phys_port_cnt:                  1
        Device ports:
                port:   1
                        state:                  PORT_ACTIVE (4)
                        max_mtu:                4096 (5)
                        active_mtu:             1024 (3)
                        sm_lid:                 0
                        port_lid:               0
                        port_lmc:               0x00
                        link_layer:             Ethernet

hca_id: mlx5_0
        transport:                      InfiniBand (0)
        fw_ver:                         12.14.1100
        node_guid:                      248a:0703:003f:1836
        sys_image_guid:                 248a:0703:003f:1836
        vendor_id:                      0x02c9
        vendor_part_id:                 4115
        hw_ver:                         0x0
        board_id:                       MT_2190110032
        phys_port_cnt:                  1
        Device ports:
                port:   1
                        state:                  PORT_ACTIVE (4)
                        max_mtu:                4096 (5)
                        active_mtu:             4096 (5)
                        sm_lid:                 1
                        port_lid:               2
                        port_lmc:               0x00
                        link_layer:             InfiniBand
```

The ud_mlx5/uct_flush_test.am_zcopy_flush_ep_no_comp/0 test hangs as well.
MLNX_OFED_LINUX-3.3-1.0.4.0

### PR DESCRIPTION
**Pull request by yosefe**
Fix #996

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2451/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1793/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by yosefe** [comment]
@alex-mikheev pls take a look

File: src/uct/ib/mlx5/ib_mlx5.h [hunk 1]
```diff
@@ -379,22 +379,29 @@ uct_ib_mlx5_ep_set_rdma_seg(struct mlx5_wqe_raddr_seg *raddr, uint64_t rdma_radd
...
     mlx5_av_base(&seg->av)->dqp_dct        = mlx5_av_base(av)->dqp_dct;
-/*  No need to fill grh
-    seg->av.grh_sec.tclass      = av->grh_sec.tclass;
-    seg->av.grh_sec.hop_limit   = av->grh_sec.hop_limit;
-    seg->av.grh_sec.grh_gid_fl  = av->grh_sec.grh_gid_fl;
-*/
-    mlx5_av_grh(&seg->av)->grh_gid_fl  = 0;
+
+    if (is_global) {
+#if HAVE_STRUCT_MLX5_GRH_AV_RMAC
```
Comment (alex-mikheev) [inline src/uct/ib/mlx5/ib_mlx5.h]: i think it is better to have separate function that fills av grh.

**Pull request merged**